sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -242,8 +243,34 @@ export class DefaultAzureCredential extends ChainedTokenCredential {
       createDefaultAzureDeveloperCliCredential,
     ];
 
-    // DefaultCredential constructors should not throw, instead throwing on getToken() which is handled by ChainedTokenCredential.
+    // If AZURE_TOKEN_CREDENTIALS is set, use it to determine which credentials to use.
+    // The value of AZURE_TOKEN_CREDENTIALS should be either "development" or "production".
+    if (process.env.AZURE_TOKEN_CREDENTIALS) {
+      if (process.env.AZURE_TOKEN_CREDENTIALS.toLowerCase() === "production") {
+        // If AZURE_TOKEN_CREDENTIALS is set to "production", use the production credential chain.
+        credentialFunctions = [
+          createEnvironmentCredential,
+          createDefaultWorkloadIdentityCredential,
+          createDefaultManagedIdentityCredential,
+        ];
+      } else if (process.env.AZURE_TOKEN_CREDENTIALS.toLowerCase() === "development") {
```

Comment:
```suggestion
      } else if (process.env.AZURE_TOKEN_CREDENTIALS.toLowerCase() === "dev") {
```

Reply:
Same suggestion here about trimming the env var value